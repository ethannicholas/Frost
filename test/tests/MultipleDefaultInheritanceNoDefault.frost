-- without the @default annotations, at one point the methods were bypassing the normal @default
-- method filtering and we ended up creating a Methods object with duplicate signatures, resulting
-- in an assertion failure.
interface Example1 {
    method defaultMethod() {
        Console.printLine("this method has an implementation!")
    }
}

interface Example2 {
    method defaultMethod() {
        Console.printLine("this method also has an implementation!")
    }
}

class Foo : Example2, Example1 {
}

method main() {
    Foo().defaultMethod()
}