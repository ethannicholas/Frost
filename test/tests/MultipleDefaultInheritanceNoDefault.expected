MultipleDefaultInheritanceNoDefault.frost:5:5: error: non-abstract interface method 'defaultMethod' must be annotated with '@default'
MultipleDefaultInheritanceNoDefault.frost:11:5: error: non-abstract interface method 'defaultMethod' must be annotated with '@default'
2 errors
